cmd/cosign/cli/verify/verify_blob.go

```diff
@@ -161,44 +162,69 @@ func VerifyBlobCmd(ctx context.Context, ko sign.KeyOpts, certRef, certEmail, cer
 		if len(uuids) == 0 {
 			return errors.New("could not find a tlog entry for provided blob")
 		}
+		return verifyUUIDs(ctx, ko, rClient, certEmail, certOidcIssuer, sig, b64sig, uuids, blobBytes)
+	}
+
+	// Use the DSSE verifier if the payload is a DSSE with the In-Toto format.
+	if isIntotoDSSE(blobBytes) {
+		verifier = dsse.WrapVerifier(verifier)
+	}
+
+	// verify the signature
+	if err := verifier.VerifySignature(bytes.NewReader([]byte(sig)), bytes.NewReader(blobBytes)); err != nil {
+		return err
+	}
+
+	// verify the rekor entry
+	if err := verifyRekorEntry(ctx, ko, verifier, cert, b64sig, blobBytes); err != nil {
+		return err
+	}
+
+	fmt.Fprintln(os.Stderr, "Verified OK")
+	return nil
+}
 
-		tlogEntry, err := cosign.GetTlogEntry(ctx, rClient, uuids[0])
+func verifyUUIDs(ctx context.Context, ko sign.KeyOpts, rClient *client.Rekor, certEmail, certOidcIssuer, sig, b64sig string, uuids []string, blobBytes []byte) error {
+	var validUUID bool
+	for _, u := range uuids {
+		tlogEntry, err := cosign.GetTlogEntry(ctx, rClient, u)
 		if err != nil {
-			return err
+			continue
 		}
 
 		certs, err := extractCerts(tlogEntry)
 		if err != nil {
-			return err
+			continue
 		}
 
 		co := &cosign.CheckOpts{
 			RootCerts:      fulcio.GetRoots(),
 			CertEmail:      certEmail,
 			CertOidcIssuer: certOidcIssuer,
 		}
-		cert = certs[0]
-		verifier, err = cosign.ValidateAndUnpackCert(cert, co)
+		cert := certs[0]
+		verifier, err := cosign.ValidateAndUnpackCert(cert, co)
 		if err != nil {
-			return err
+			continue
+		}
+		// Use the DSSE verifier if the payload is a DSSE with the In-Toto format.
+		if isIntotoDSSE(blobBytes) {
+			verifier = dsse.WrapVerifier(verifier)
+		}
+		// verify the signature
+		if err := verifier.VerifySignature(bytes.NewReader([]byte(sig)), bytes.NewReader(blobBytes)); err != nil {
+			continue
 		}
-	}
-
-	// Use the DSSE verifier if the payload is a DSSE with the In-Toto format.
-	if isIntotoDSSE(blobBytes) {
-		verifier = dsse.WrapVerifier(verifier)
-	}
 
-	// verify the signature
-	if err := verifier.VerifySignature(bytes.NewReader([]byte(sig)), bytes.NewReader(blobBytes)); err != nil {
-		return err
+		// verify the rekor entry
+		if err := verifyRekorEntry(ctx, ko, verifier, cert, b64sig, blobBytes); err != nil {
+			continue
+		}
+		validUUID = true
 	}
-
-	// verify the rekor entry
-	if err := verifyRekorEntry(ctx, ko, verifier, cert, b64sig, blobBytes); err != nil {
-		return err
+	if !validUUID {
+		return errors.New("could not find a valid tlog entry for provided blob")
```

Comment:
I can imagine this being really difficult to debug. Maybe: "could not find....blob. found X, none verified successfully"